core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { NgClass, NgIf } from '@angular/common';
+import { Component, DebugElement } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { ButtonModule } from 'primeng/button';
+import { CardModule } from 'primeng/card';
+import { ScrollPanelModule } from 'primeng/scrollpanel';
+
+import {
+    TemplateBuilderBoxComponent,
+    TemplateBuilderBoxSize
+} from './template-builder-box.component';
+
+@Component({
+    selector: 'dotcms-host-component',
+    template: ` <dotcms-template-builder-box [size]="size"> </dotcms-template-builder-box>`
+})
+class TemplateBuilderBoxHostComponent {
+    size = 'large';
+}
+
+describe('TemplateBuilderBoxComponent', () => {
+    let component: TemplateBuilderBoxHostComponent;
+    let hostFixture: ComponentFixture<TemplateBuilderBoxHostComponent>;
+    let de: DebugElement;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [
+                TemplateBuilderBoxComponent,
+                NgClass,
+                NgIf,
+                ButtonModule,
+                CardModule,
+                ScrollPanelModule
+            ],
+            declarations: [TemplateBuilderBoxHostComponent]
+        }).compileComponents();
+
+        hostFixture = TestBed.createComponent(TemplateBuilderBoxHostComponent);
+        component = hostFixture.componentInstance;
+        de = hostFixture.debugElement;
+        hostFixture.detectChanges();
+    });
+
+    it('should create the component', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should render with default size', () => {
+        expect(de.nativeElement.querySelector('.template-builder-box__large')).toBeTruthy();
+    });
+
+    it('should render with medium size and update the class', async () => {
```

Comment:
do we need this async?

Reply:
We might want to use Spectator here: https://ngneat.github.io/spectator/

@oidacra and @hmoreras are already using it in the Experiment portlet and one of the cool thing is that you can create host parent components.

Reply:
this file has an example `dot-experiment-options.component.spec.ts`